### DISCUSSION TITLE
Improvements

### DISCUSSION THREAD

**Discussion by rpivo** [rpivo/react-automation-profiler#62] [Ideas]
Here we can discuss general improvements that can be made on react-automation-profiler.

**Comment by rpivo** [comment]
Some things I'm looking into in the next week:
- try fixing issue #57 if it's still occurring.
- update dependencies.
- expand on example folder. This example should make use of the flags and features and highlight a variety of scenarios.

Hopefully soon:
- adding tests to prevent regressions

cc @AlithAnar

**Comment by AlithAnar** [comment]
I've got other ideas for the improvement/enhancement:

1. Replace `printMessage` with a custom logger or external package e.g. [winston](https://www.npmjs.com/package/winston). The goal is to simplify logging API (logger.log, logger.error, logger.info etc.) and provide multi-level logging.

2. Introduce programmatic API for the profiler to be used inside the external code: let's say that I want to build a centralized performance metrics manager for our main application, I could execute the automation inside the code, retrieve raw output, post-process it, and send to external metrics system. 

```TSX
import Profiler from 'react-automation-profiler'


async function myRepoProfilingFn() {
    const options = { ... }
    const results = await Profiler.automation(options, steps)
    / * --- puppeter magic is hapening --- */
    const postProcessedResults = someParsingFunction(results)
    externalService.send(postProcessedResults)
}
```
